lib/systemd/user-level-services.nix

```diff
@@ -0,0 +1,88 @@
+{ nixpkgs
+, system ? "x86_64-linux"
+, pkgs ? nixpkgs.legacyPackages.${system}, ... }:
+let
+  mkVM = moduleConfig: serviceConfig: nixpkgs.lib.nixosSystem {
+    inherit system;
+    modules = [ ({ ... }: {
+
+      imports = [ moduleConfig.module ];
+
+      services.${moduleConfig.nixOsServiceName} = {
+        enable = true;
+      } // serviceConfig;
+
+    }) ];
+  };
+
+in {
+  mkActivateScript = activateScriptName:
+      { module, nixOsServiceName, primaryService, primaryPortOptionName, auxiliaryServices, oneShotServices}@moduleConfig:
+      { backupAction ? "", restoreAction ? ""}:
+      serviceConfig:
+    let
+      vm = mkVM moduleConfig serviceConfig;
+      wantedServices = [primaryService] ++ auxiliaryServices;
```

Comment:
This service separation doesn't look very proper to me, at least with the current naming. Oneshot services may be required to be started on reboot, e.g. consider a oneshot service that performs a DB migration prior to backend start.
While the actual separation services on those that need to be started after `default.target` and those that aren't is correct, I don't think that it should be deduced by anything other than service `wantedBy` value, i.e. it shouldn't be necessary to enumerate services manually

Reply:
Done.
`oneShotServices` has been unified with `auxiliaryServices`, we decide to make a link or not on the presence of `wantedBy`